repository/src/main/java/ru/karalionak/rest/dao/impl/CertificateDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+package ru.karalionak.rest.dao.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Component;
+import ru.karalionak.rest.dao.AbstractDao;
+import ru.karalionak.rest.dao.CertificateDao;
+import ru.karalionak.rest.entity.Certificate;
+import ru.karalionak.rest.query.QueryHelper;
+import ru.karalionak.rest.query.SortParamContext;
+
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class CertificateDaoImpl extends AbstractDao<Certificate> implements CertificateDao {
+    private final JdbcTemplate jdbcTemplate;
+    private final QueryHelper queryHelper;
+    private static final RowMapper<Certificate> certificateRowMapper = (ResultSet rs, int rowNum) -> new Certificate(
+            rs.getLong("id"),
+            rs.getString("certificate_name"),
+            rs.getString("description"),
+            rs.getBigDecimal("price"),
+            rs.getInt("duration"),
+            rs.getTimestamp("create_date").toLocalDateTime(),
+            rs.getTimestamp("last_update_date").toLocalDateTime()
+    );
+    private final static String TABLE_NAME = "gift_certificate";
+
+    @Autowired
+    public CertificateDaoImpl(JdbcTemplate jdbcTemplate, QueryHelper queryHelper) {
+        super(jdbcTemplate, certificateRowMapper, TABLE_NAME);
+        this.jdbcTemplate = jdbcTemplate;
+        this.queryHelper = queryHelper;
+    }
+
+    @Override
+    public void create(Certificate certificate) {
+        jdbcTemplate.update("INSERT INTO gift_certificate(certificate_name, description, price, duration) " +
+                        "VALUES(?, ?, ?, ?)", certificate.getName(), certificate.getDescription(),
+                certificate.getPrice(), certificate.getDuration());
+    }
+
+    @Override
+    public void update(long id, Map<String, Object> updatedColumns) {
+        if (!updatedColumns.isEmpty()) {
+            StringBuilder query = new StringBuilder("UPDATE gift_certificate SET ");
+            query.append(queryHelper.buildUpdateQuery(updatedColumns.keySet()))
+                    .append(" ")
+                    .append("WHERE id=?");
+            jdbcTemplate.update(query.toString(), updatedColumns.values().toArray(), id);
+        }
+    }
+
+    @Override
+    public void createCertificateTagRef(long certificateId, long tagId) {
+        jdbcTemplate.update("INSERT INTO gift_certificate_tag(certificate_id, tag_id) VALUES(?,?)",
+                certificateId, tagId);
+    }
+
+
+    @Override
+    public List<Long> getTagIdsByCertificateId(long certificateId) {
```

Comment:
It can be implemented in TagDao. And should looks like List<Tag> findTagsByCertificateId(long certificateId)